feat: 공통 레이아웃(Header, Sidebar) 및 SidebarTab 컴포넌트 구현

MUNJEONGJUN opened the pull request:
- 공통 레이아웃(Header, Sidebar) 구현 및 App > layout.tsx 파일에 적용
- 프로젝트 진행 단계별 목록을 보여주는 SidebarTab 컴포넌트 추가

### 📄 Description of the PR

- 기능 설명 : 공통 레이아웃 구현
- Related to #3 공통 레이아웃 

### 🔧 What has been changed?

<!-- 이번 PR에서의 변경점 (- 코드 변경, 기능 추가, 버그 수정 등.)-->

- 공통 레이아웃(Header, Sidebar) 구현
- 프로젝트 진행 단계(진행 중/완료) 별 목록을 보여주는 SidebarTab 컴포넌트 구현
- (기타) 페이지 전환용 Drawer 컴포넌트 구현

### 📸 Screenshots / GIFs (if applicable)

<!-- 가능하다면 스크린샷이나 GIF를 첨부해주세요 -->

<img width="959" alt="image" src="https://github.com/user-attachments/assets/a77038b8-78cd-40e5-aa26-2d31f7fba408" />
<img width="959" alt="image" src="https://github.com/user-attachments/assets/03da1f06-67eb-458a-a1f3-4541bd5ea69c" />
<img width="953" alt="image" src="https://github.com/user-attachments/assets/649f99b3-3c3b-45f8-87fe-7a9653eeaee7" />

### ⚠️ Precaution & Known issues

<!-- 리뷰할 때 고려해야 할 부분, 수정이 필요한 부분, 아직 해결되지 않는 문제, 개선해야 할 사항 -->

App > layout.tsx 파일
- 공통 레이아웃 적용 (Header, Sidebar)

components > common 폴더 (공통 컴포넌트)
- common 폴더 밑에 SidebarTab 컴포넌트 추가

components > layouts 폴더 생성 (공통 레이아웃)
- layouts 폴더 밑에 Header, Sidebar, Drawer 추가

data 폴더 (MOCK 데이터)
- data 폴더 밑에 users_mock_data.json 파일 추가 (회원 data)

components > pages 폴더
- 불필요한 페이지 제거

### ✅ Checklist

- [ ] UI 브랜치 같이 확인해서 이슈없는지 확인해보기
- [ ] 함수 이름, 변수 이름만 봐도 어떤 기능을 하는지 파악할 수 있는지 (선언적인 코드인지 확인)

MUNJEONGJUN committed:
feat: 공통 레이아웃(Header, Sidebar) 및 SidebarTab 컴포넌트 구현

- 공통 레이아웃(Header, Sidebar) 구현 및 App > layout.tsx 파일에 적용
- 프로젝트 진행 단계별 목록을 보여주는 SidebarTab 컴포넌트 추가

Sang-Ho-Jeon commented on package-lock.json:
```diff
@@ -5449,7 +5449,6 @@
       "version": "5.4.0",
       "resolved": "https://registry.npmjs.org/react-icons/-/react-icons-5.4.0.tgz",
       "integrity": "sha512-7eltJxgVt7X64oHh6wSWNwwbKTCtMfK35hcjvJS0yxEAhPM8oUKdS3+kqaW1vicIltw+kR2unHaa12S9pPALoQ==",
-      "license": "MIT",
```
이 구문은 삭제한 이유가 뭔가요?

MUNJEONGJUN replied:
직접 삭제하지 않았습니다!
처음 겪어보는 이슈인데 package-lock.json 파일에 라이센스 부분이 자동으로 삭제되는 경우가 있나요?

해당 코드는 react-icons 부분이며,
공식 사이트에 확인해보니 5.4.0 버전은 라이센스가 MIT 로 확인됩니다.

**코드 실행에 문제는 없을 것으로 보이니,
삭제된 부분은 추가하고 추후에 같은 이슈가 재발하면 그 때 다시 논의하면 어떨까요?**

cf) react-icons 공식 문서
https://www.npmjs.com/package/react-icons

cf) package.json 파일과 package-lock.json 파일
- package.json : 패키지 설치, 배포, 의존성 관련 정보
- package-lock.json : 패키지 버전, 의존성 트리

MUNJEONGJUN replied:
(논의 완료)
라이센스 MIT 추가하여 다시 PUSH 하겠습니다.

Sang-Ho-Jeon commented on components/layouts/Sidebar.tsx:
```diff
@@ -0,0 +1,51 @@
+        {/* Segmented Control */}
+        <SegmentedControl
+          value={value}
+          onValueChange={e => {
```
이벤트 객체 타입을 지정해주세요

MUNJEONGJUN replied:
onValueChange 말씀이시죠?

Chakra UI 에서 정의된 객체로 타입이 이미 정의되어 있습니다. 저희가 타입을 추가적으로 지정하지 않아도 될 것 같은데 의견 부탁드립니다. 관련 코드 첨부드립니다.

<index.d.ts>
`interface ValueChangeDetails {
    value: string;
}`

`    onValueChange?(details: ValueChangeDetails): void;
`

MUNJEONGJUN replied:
(논의 완료)
<SegmentedControl> 사용 전에 아래 구문을 통해 설치해주시면 자동 타입 지정될 것으로 보입니다.

(If you don't already have the snippet, run the following command to add the segmented-control snippet)

npx @chakra-ui/cli snippet add segmented-control

Sang-Ho-Jeon commented on components/layouts/Sidebar.tsx:
```diff
@@ -0,0 +1,51 @@
+  return (
+    <Flex flexDirection="column" gap="1" bg="gray.500">
+      <Box width="100%" height="100vh" p={1}>
+        {/* Avatar */}
```
아바타 컴포넌트를 따로 분리시켜서 헤더의 프로필과 공통 컴포넌트로 사용하면 좋을거같아요

MUNJEONGJUN replied:
(논의 완료)
- 아바타 컴포넌트 분리 완료
- 파일 위치 변경 (components > common > Profile.tsx)

Sang-Ho-Jeon commented on components/common/SidebarTab.tsx:
```diff
@@ -0,0 +1,21 @@
+import Data from "@/data/projects_mock_data.json";
+import Link from "next/link";
+
+export default function SidebarTab({ props }) {
```
props의 타입을 지정해주세요. 네이밍도 변경하면 좋을거같아요.

MUNJEONGJUN replied:
(논의 완료)
- props 타입 지정 완료 (string)
- props 변수명 변경 완료 (projectStatus)

Sang-Ho-Jeon commented on app/page.tsx:
공통 레이아웃을 분리시키는거 좋습니다!

Sang-Ho-Jeon commented on app/layout.tsx:
컴포넌트분리 잘 하신것같습니다. 굿!

Sang-Ho-Jeon reviewed (changes requested):
사소한 부분 빼고는 컴포넌트 분리가 적절히 잘 이루어진 것 같습니다.

MUNJEONGJUN committed:
refactor: 공통 레이아웃 리뷰 반영

1. (package-lock.json) react-icons 라이센스 삭제 취소
2. (sidebarTab) props, 변수명 지정
3. (Profile.tsx) 아바타 컴포넌트 별도 분리 후 공통 컴포넌트로 재구현

Sang-Ho-Jeon reviewed (approved):
수정사항이 잘 반영되었네요.
